pkg/cmd/provisioning/alibabacloud/attach-ram-policy.go

```diff
@@ -0,0 +1,293 @@
+}
+
+// writeCredReqSecret will take a credentialsRequest and return
+// a Secret with the AK/SK of a ram user who has grant component permission defined in credentialsRequest.
+func writeCredReqSecret(cr *credreqv1.CredentialsRequest, componenntSecretInfo ComponenntSecretInfo) error {
+	manifestsDir := filepath.Join(componenntSecretInfo.targetDir, provisioning.ManifestsDirName)
+
+	fileName := fmt.Sprintf("%s-%s-credentials.yaml", cr.Spec.SecretRef.Namespace, cr.Spec.SecretRef.Name)
+	filePath := filepath.Join(manifestsDir, fileName)
+
+	fileData := fmt.Sprintf(secretManifestsTemplate, componenntSecretInfo.componentAccessKeyId, componenntSecretInfo.componentAccessKeySecret, cr.Spec.SecretRef.Name, cr.Spec.SecretRef.Namespace)
+
+	if err := ioutil.WriteFile(filePath, []byte(fileData), 0600); err != nil {
+		return errors.Wrap(err, "Failed to save Secret file")
+	}
+
+	log.Printf("Saved credentials configuration to: %s", filePath)
+
+	return nil
+}
+
+// initEnvForAttachRAMPolicyCmd will ensure the env CCOCTL_ALIBABA_COMPONENT_ACCESS_KEY_ID and CCOCTL_ALIBABA_COMPONENT_ACCESS_KEY_ID defined
+// and ensure destination directory is ready to receive the generated
+// files, and will create the directory if necessary.
+func initEnvForAttachRAMPolicyCmd(cmd *cobra.Command, args []string) {
+	//ensure env CCOCTL_ALIBABA_COMPONENT_ACCESS_KEY_ID and CCOCTL_ALIBABA_COMPONENT_ACCESS_KEY_ID set
+	//for Alibaba Cloud components permission binding
+	componentAk := os.Getenv(componentAccessKeyId)
+	componentSk := os.Getenv(componentAccessKeySecret)
+	if componentAk == "" || componentSk == "" {
+		log.Fatalf("Empty value of CCOCTL_ALIBABA_COMPONENT_ACCESS_KEY_ID or CCOCTL_ALIBABA_COMPONENT_ACCESS_KEY_SECRET")
+	}
+	AttachRAMPolicyOpts.ComponentAccessKeyId = componentAk
+	AttachRAMPolicyOpts.ComponentAccessKeySecret = componentSk
+	//call sts GetCallerIdentity for the username of given component ak/sk
+	stsClient, err := sts.NewClientWithAccessKey(AttachRAMPolicyOpts.Region, componentAk, componentSk)
+	if err != nil {
+		log.Fatalf("Failed to init sts client from ram username: %++v", refineMissingRegionIdErr(err))
+	}
+	req := sts.CreateGetCallerIdentityRequest()
+	req.Scheme = apiRequestScheme
+	res, err := stsClient.GetCallerIdentity(req)
+	if err != nil {
+		log.Fatalf("Failed to get caller identity: %++v", err)
+	}
+	userName, err := getRAMUserName(res.Arn)
+	if err != nil {
+		log.Fatalf("Failed to get ram user name: %++v", err)
+	}
+	AttachRAMPolicyOpts.UserName = userName
+	if AttachRAMPolicyOpts.TargetDir == "" {
+		pwd, err := os.Getwd()
+		if err != nil {
+			log.Fatalf("Failed to get current directory: %s", err)
+		}
+
+		AttachRAMPolicyOpts.TargetDir = pwd
+	}
+
+	fPath, err := filepath.Abs(AttachRAMPolicyOpts.TargetDir)
+	if err != nil {
+		log.Fatalf("Failed to resolve full path: %s", err)
+	}
+
+	// create target dir if necessary
+	err = provisioning.EnsureDir(fPath)
+	if err != nil {
+		log.Fatalf("failed to create target directory at %s", fPath)
```

Comment:
```suggestion
		log.Fatalf("Failed to create target directory at %s", fPath)
```
Can you make every log message start with `uppercase`. Just to be consistent